lib/comments.ts

```diff
@@ -0,0 +1,51 @@
+import supabase from '#/lib/supabase-browser';
+import { ok } from 'assert';
+import type { Database } from './database.types';
+import type { User } from './users';
+
+export const getComments = async ({
+  postId,
+  nodeId,
+}: {
+  postId?: number | null;
+  nodeId?: number | null;
+}): Promise<Comment[]> => {
```

Comment:
If it's `else-if` (only `postId` or `nodeId`, never both), you can add a union here:

```suggestion
}: {
  postId?: number | null;
} | {
  nodeId?: number | null;
}): Promise<Comment[]> => {
```

And it looks like you require one of them. This will make the `else` case with the Error obsolete:

```suggestion
}: {
  postId: number | null;
} | {
  nodeId: number | null;
}): Promise<Comment[]> => {
```

Now it's required to set `postId` or `nodeId`.

If you do that, you need to do the same for `useComments`

Reply:
Oh that's pretty nice, totally forgot that we were able to do that!